cgen convert empty interface to array

### Describe the bug

While attempting to minimally replicate #26760 I ran into this:


### Reproduction Steps

```
module main

import arrays

interface Value {}

fn called(v ...Value) {
	println(v)
}

fn caller() {
	mut v := []Value{}
	v = arrays.concat(v, 'brother')
	println(v)
	called(v) // accidentally put v instead of ...v
}

fn main() {
	caller()
}
```

### Expected Behavior

runs

### Current Behavior

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/26760.01KMDTTJDHME2JVTBA9PMGPEKH.tmp.c:5587: error: cannot convert 'struct main__Value' to 'struct array'
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.5.1 58f1857

### Environment details (OS name and version, etc.)

```
|V full version      |V 0.5.1 58f18573557247e9c743f7d60e5b3ab783f74479
|:-------------------|:-------------------
|OS                  |linux, "openmamba release 2025.12.1 for x86_64 (rolling)"
|Processor           |8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz
|Memory              |16.27GB/31.17GB
|                    |
|V executable        |/home/einar/.local/lib64/v/v
|V last modified time|2026-03-23 10:47:07
|                    |
|V home dir          |OK, value: /home/einar/.local/lib64/v
|VMODULES            |OK, value: /home/einar/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/einar/Documents/projects/vlang/notes
|                    |
|Git version         |git version 2.53.0
|V git status        |weekly.2026.03-571-g58f18573
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.2.1 20251019
|gcc version         |gcc (GCC) 15.2.1 20251019
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.43
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2844,8 +2844,8 @@ fn (mut g Gen) call_args(node ast.CallExpr) {
 						}
 					}
 				} else {
-					// passing variadic arg to another call which expects same array type
-					if variadic_count == 1
+					// passing the current variadic arg to another call which expects the same array type
+					if variadic_count == 1 && g.is_forwarded_variadic_arg(args[arg_nr])
 						&& ((args[arg_nr].typ.has_flag(.variadic) && args[arg_nr].typ == varg_type)
 						|| (varg_type.has_flag(.variadic)
 						&& args[arg_nr].typ == varg_type.clear_flag(.variadic)
@@ -2883,6 +2883,18 @@ fn (mut g Gen) call_args(node ast.CallExpr) {
 			}
 		}
 	}
+}
+
+fn (g &Gen) is_forwarded_variadic_arg(arg ast.CallArg) bool {
+	if arg.typ.has_flag(.variadic) {
+		return true
+	}
+	if arg.expr is ast.Ident && g.cur_fn != unsafe { nil } && g.cur_fn.is_variadic
+		&& g.cur_fn.params.len > 0 && arg.expr.obj is ast.Var {
+		var_obj := arg.expr.obj as ast.Var
+		return var_obj.is_arg && arg.expr.name == g.cur_fn.params.last().name
+	}
+	return false
 }
 
 // similar to `autofree_call_pregen()` but only to to handle [keep_args_alive] for C functions

--- a/vlib/v/tests/fns/variadic_interface_array_arg_test.v
+++ b/vlib/v/tests/fns/variadic_interface_array_arg_test.v
@@ -1,0 +1,24 @@
+interface VariadicValue {}
+
+fn variadic_capture(values ...VariadicValue) []VariadicValue {
+	return values
+}
+
+fn test_variadic_interface_array_variable_as_single_arg() {
+	nested := [VariadicValue('brother')]
+	got := variadic_capture(nested)
+	assert got.len == 1
+
+	inner := got[0] as []VariadicValue
+	assert inner.len == 1
+	assert (inner[0] as string) == 'brother'
+}
+
+fn test_variadic_interface_array_literal_as_single_arg() {
+	got := variadic_capture([VariadicValue('x')])
+	assert got.len == 1
+
+	inner := got[0] as []VariadicValue
+	assert inner.len == 1
+	assert (inner[0] as string) == 'x'
+}

